### ISSUE TITLE
[Bug] `LBOptimizer.all_reduce_grads` reduces gradients of only a model, even if training several models

### ISSUE THREAD

**Issue by wkcn** [Azure/MS-AMP#62]
**What's the issue, what's expected?**:
In the function `LBOptimizer.all_reduce_grads`, the gradients of a passed model will be reduced when `model_state.ready_to_all_reduce_grads` is True, otherwise the gradients are not reduced.

When training more than one models, the gradients of the first model are reduced then `model_state.ready_to_all_reduce_grads` is set to False. However, **the gradients of the second model are not reduced** since `model_state.ready_to_all_reduce_grads` is False.

https://github.com/Azure/MS-AMP/blob/main/msamp/optim/optimizer.py#LL47C1-L55C54
```python
    def all_reduce_grads(self, model):
        """All-reduce gradients of parameters."""
        if not model_state.ready_to_all_reduce_grads:  # skip when model_state.ready_to_all_reduce_grads is False
            return
        scaling_params = [p for p in model.parameters() if isinstance(p, ScalingParameter)]
        grads = [p.grad for p in scaling_params if p.grad is not None]
        TensorDist.all_reduce_avg(grads)
        # make sure that FP8 weight gradients have been reduced.
        model_state.ready_to_all_reduce_grads = False
```

**How to reproduce it?**:
```python
model1 = Model1()
model2 = Model2()

loss = loss_func1(model1(x), target1) + loss_func2(model2(x), target2)
loss.backward()

# `model_state.ready_to_all_reduce_grads` will be set to False. The gradients of model1 are reduced,
optimizer.all_reduce_grads(model1)
# [BUG Here] `model_state.ready_to_all_reduce_grads` has been False. The gradients of model2 will not be reduced,
optimizer.all_reduce_grads(model2)
```

**Log message or shapshot?**:

**Additional information**:
